### ISSUE TITLE
Fix S2259: False positive in switch statement with conditional access operator

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonar-dotnet#1183]
### Description

False positive when used with the null-conditional operator

### Repro steps

```
void Foo(object o)
{
    switch (o?.GetHashCode())
    {
        case 1:
            o.ToString(); // False Positive
            break;
        default:
            break;
    }
}
```

If `o` is null then `o?.GetHashCode()` is null so the _switch_ statement will select the default clause.

Example in [SonarLint](https://next.sonarqube.com/sonarqube/project/issues?id=sonarlint-visualstudio&issues=AWGuYbUhaHWQ5zPscion&open=AWGuYbUhaHWQ5zPscion). 

### Related information

* SonarC# Version: 6.7

**Comment by valhristov** [comment]
The CFG for the following code:

```
void M(object o)
{
	switch (o?.GetHashCode())
	{
		case 5:
			o.ToString();
		default:
			// do nothing
	}
}
```

is the following

```
BranchBlock1(o?.GetHashCode)
	TrueSuccessor: BranchBlock2 // o == null
	FalseSuccessor: SimpleBlock1 // o != null
SimpleBlock1
	Successor: BranchBlock2
BranchBlock2(switch)
	Successor1: JumpBlock1
	Successor2: ExitBlock
JumpBlock1(case 5)
ExitBlock
```

To fix the FP, the TrueSuccessor of BranchBlock1 should be a new (different) switch block with a single connection to the default statement.

**Comment by af4jm** [comment]
another example of the conditional access operator resulting in a false positive...
```
if (invocationList?.Length <= 0)
{
    return;
}

foreach (Delegate action in invocationList) // false positive... if invocationList is null we've returned above
```

**Comment by andrei-epure-sonarsource** [comment]
fixed in #2497
